Well, after the `RestoreEntity` refactor, now I am having a few `Too many ancestors` pylint errors:
```
************* Module homeassistant.components.switch.rflink
homeassistant/components/switch/rflink.py:70:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
************* Module homeassistant.components.light.rflink
homeassistant/components/light/rflink.py:152:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
homeassistant/components/light/rflink.py:158:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
homeassistant/components/light/rflink.py:192:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
homeassistant/components/light/rflink.py:244:0: R0901: Too many ancestors (8/7) (too-many-ancestors)
```

For all the other Travis errors, I will take a look when I have a break.

I think that refactor the inheritance for RFLink classes wil be a delicate task that I do not know if I can/must assume.
Code 'owner' has commented on several occasions that he can not devote time to it.

I will see what can I do.

--- a/homeassistant/components/light/rflink.py
+++ b/homeassistant/components/light/rflink.py
@@ -18,7 +18,6 @@ from homeassistant.components.rflink import (
     EVENT_KEY_COMMAND, EVENT_KEY_ID, SwitchableRflinkDevice,
     remove_deprecated)
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.restore_state import RestoreEntity
 from homeassistant.const import (CONF_NAME, CONF_TYPE)
 
 DEPENDENCIES = ['rflink']
@@ -155,7 +154,7 @@ class RflinkLight(SwitchableRflinkDevice, Light):
     pass
 
 
-class DimmableRflinkLight(SwitchableRflinkDevice, Light, RestoreEntity):
+class DimmableRflinkLight(SwitchableRflinkDevice, Light):
     """Rflink light device that support dimming."""
 
     _brightness = 255
@@ -189,7 +188,7 @@ class DimmableRflinkLight(SwitchableRflinkDevice, Light, RestoreEntity):
         return SUPPORT_BRIGHTNESS
 
 
-class HybridRflinkLight(SwitchableRflinkDevice, Light, RestoreEntity):
+class HybridRflinkLight(SwitchableRflinkDevice, Light):
     """Rflink light device that sends out both dim and on/off commands.
 
     Used for protocols which support lights that are not exclusively on/off

--- a/homeassistant/components/light/rflink.py
+++ b/homeassistant/components/light/rflink.py
@@ -210,7 +210,7 @@ class HybridRflinkLight(SwitchableRflinkDevice, Light, RestoreEntity):
         """Restore RFLink light brightness attribute."""
         await super().async_added_to_hass()
 
-        old_state = await async_get_last_state(self.hass, self.entity_id)
+        old_state = await self.async_get_last_state()
         if old_state is not None and \
                 old_state.attributes.get(ATTR_BRIGHTNESS) is not None:
             self._brightness = int(old_state.attributes.get(ATTR_BRIGHTNESS))

--- a/homeassistant/components/light/rflink.py
+++ b/homeassistant/components/light/rflink.py
@@ -148,12 +148,14 @@ async def async_setup_platform(hass, config, async_add_entities,
         hass.data[DATA_DEVICE_REGISTER][EVENT_KEY_COMMAND] = add_new_device
 
 
+# pylint: disable=too-many-ancestors
 class RflinkLight(SwitchableRflinkDevice, Light):
     """Representation of a Rflink light."""
 
     pass
 
 
+# pylint: disable=too-many-ancestors
 class DimmableRflinkLight(SwitchableRflinkDevice, Light):
     """Rflink light device that support dimming."""
 
@@ -188,6 +190,7 @@ class DimmableRflinkLight(SwitchableRflinkDevice, Light):
         return SUPPORT_BRIGHTNESS
 
 
+# pylint: disable=too-many-ancestors
 class HybridRflinkLight(SwitchableRflinkDevice, Light):
     """Rflink light device that sends out both dim and on/off commands.
 
@@ -240,6 +243,7 @@ class HybridRflinkLight(SwitchableRflinkDevice, Light):
         return SUPPORT_BRIGHTNESS
 
 
+# pylint: disable=too-many-ancestors
 class ToggleRflinkLight(SwitchableRflinkDevice, Light):
     """Rflink light device which sends out only 'on' commands.
 

--- a/homeassistant/components/switch/rflink.py
+++ b/homeassistant/components/switch/rflink.py
@@ -67,6 +67,7 @@ async def async_setup_platform(hass, config, async_add_entities,
     async_add_entities(devices_from_config(config))
 
 
+# pylint: disable=too-many-ancestors
 class RflinkSwitch(SwitchableRflinkDevice, SwitchDevice):
     """Representation of a Rflink switch."""
 

